src/domains/chart/components/snapshot-loader.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from "react"
+
+import { MS_IN_SECOND } from "utils"
+import { serverDefault } from "utils/server-detection"
+import { selectIsSnapshotMode, selectSnapshotOptions } from "domains/dashboard/selectors"
+import { selectGlobalPanAndZoom } from "domains/global/selectors"
+import { useDispatch, useSelector } from "store/redux-separate-context"
+
+import { Attributes } from "../utils/transformDataAttributes"
+import { fetchDataForSnapshotAction } from "../actions"
+import { chartLibrariesSettings } from "../utils/chartLibrariesSettings"
+import { getChartURLOptions } from "../utils/get-chart-url-options"
+
+interface SnapshotLoaderProps {
+  attributes: Attributes
+  chartUuid: string
+}
+const SnapshotLoader = ({
+  attributes,
+  chartUuid,
+}: SnapshotLoaderProps) => {
+  const host = attributes.host || serverDefault
+  const { snapshotDataPoints } = useSelector(selectSnapshotOptions)
+  const group = attributes.method || window.NETDATA.chartDefaults.method
+  const { chartLibrary } = attributes
+  const chartSettings = chartLibrariesSettings[chartLibrary]
+
+  const globalPanAndZoom = useSelector(selectGlobalPanAndZoom)
+  const after = (globalPanAndZoom as { after: number, before: number }).after / MS_IN_SECOND
+  const before = (globalPanAndZoom as { after: number, before: number }).before / MS_IN_SECOND
```

Comment:
`{ after: number, before: number }`

could be `type RangeT = { after: number, before: number }`

Reply:
done